src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -124,10 +124,16 @@ internal virtual async Task FlushLogs(CloudTable table = null)
                 return;
             }
 
-            table = table ?? GetDiagnosticEventsTable();
-
             try
             {
+                table = table ?? GetDiagnosticEventsTable();
+
+                if (table == null)
+                {
+                    _logger.LogError("Unable to get table reference. Aborting write operation");
```

Comment:
I don't think we want to see a persistent error on each time the flush timer fires. Instead, I don't think we should be running the flush timer at all if we don't have storage, and to Pragna's point, we can probably even disable the diagnostic logger even earlier so it isn't even operating?

Reply:
That makes sense. disabling the diagnostic event logger makes sense, other loggers will catch and log the statements anyway.

Reply:
Instead of disabling the feature and pushing knowledge about the repository implementation to other components, I would recommend the approach I’ve suggested as that enables us to provide an efficient implementation and provides separation of concerns.